cpp/src/Glacier2Lib/NullPermissionsVerifier.cpp

```diff
@@ -131,9 +54,64 @@ namespace Glacier2Internal
 {
 
 void
-setupNullPermissionsVerifier(const CommunicatorPtr& communicator, const string& category, const vector<string>& props)
+setupNullPermissionsVerifier(
+    const CommunicatorPtr& communicator,
+    const string& category,
+    const vector<string>& permissionsVerifierPropertyNames)
 {
-    Init init(communicator, category, props);
+    const Ice::Identity nullPermissionsVerifierId { "NullPermissionsVerifier", category };
+    const Ice::Identity nullSSLPermissionsVerifierId { "NullSSLPermissionsVerifier",  category };
+
+    Ice::PropertiesPtr properties = communicator->getProperties();
+    ObjectAdapterPtr adapter;
+
+    shared_ptr<Glacier2::PermissionsVerifier> nullPermissionsVerifier;
+    shared_ptr<Glacier2::SSLPermissionsVerifier> nullSSLPermissionsVerifier;
+
+    for (const auto& propertyName : permissionsVerifierPropertyNames)
+    {
+        string val = properties->getProperty(propertyName);
+        if (!val.empty())
+        {
+            try
+            {
+                ObjectPrx prx(communicator, val);
+                if (prx->ice_getIdentity() == nullPermissionsVerifierId && nullPermissionsVerifier == nullptr)
+                {
+                    adapter = createObjectAdapter(communicator, adapter);
+                    nullPermissionsVerifier = make_shared<NullPermissionsVerifier>();
```

Comment:
This is continuously re-creating `nullPermissionsVerifier`. Why not just create it once and share it access all implementations.

Reply:
This doesn't re-create it. It checks `&& nullPermissionsVerifier == nullptr` to avoid re-creating.

Reply:
I missed this part of the check,